collect_app/src/main/java/org/odk/collect/android/adapters/InstanceListCursorAdapter.java

```diff
@@ -163,6 +164,6 @@ public static int getFormStateImageResourceIdForStatus(String formStatus) {
                 return R.drawable.ic_form_state_submission_failed;
         }
 
-        return -1;
+        throw new IllegalArgumentException();
```

Comment:
Returning `-1` here causes a crash further down stream that's harder to understand.